api/axis.py

```diff
@@ -435,6 +435,19 @@ def fulfill(self, patron, pin, licensepool, internal_format, **kwargs):
             )
 
         identifier = licensepool.identifier
+        if isinstance(patron, Patron):
+            # Check if the Patron's Loan has
+            # a transaction ID as an external_identifier
+            loan, _ = licensepool.loan_to(patron)
```

Comment:
This will create a local Loan object starting at the current moment, even if no loan actually exists. Instead, use `get_one(_db, Loan, patron=patron, licensepool=licensepool)` to fetch a loan only if it already exists on our side. If it doesn't, then rely on the availability call to either find it (if it was somehow created on the Axis side and we didn't hear about it) or raise an exception.

Reply:
I've updated this to use `get_one` instead of `licensepool.loan_to()`, definitely don't want to be creating a new loan here.